src/tyr/height_serializer.cc

```diff
@@ -1,60 +1,80 @@
-                                      const double no_data_value) {
-  auto array = json::array({});
+void serialize_range_height(rapidjson::writer_wrapper_t& writer,
+                            const std::vector<double>& ranges,
+                            const std::vector<double>& heights,
+                            const uint32_t precision,
+                            const double no_data_value) {
+  writer.start_array("range_height");
+  // precisoin is sent with value 0 which breaks the condition: maxDecimalPlaces >= 1 and causes an
+  // error
+  writer.set_precision(std::max(precision, 1u));
   // for each posting
   auto range = ranges.cbegin();
 
   for (const auto height : heights) {
-    auto element = json::array({json::fixed_t{*range, 0}});
+    writer.start_array();
+
+    if (precision == 0) {
+      // cast to integer after rounding instead of changing tests in valhalla/test/skadi_service.cc
+      writer(static_cast<int64_t>(std::round(*range)));
+    } else {
+      writer(*range);
+    }
+
     if (height == no_data_value) {
-      element->push_back(nullptr);
+      writer(nullptr);
+    } else if (precision == 0) {
+      // cast to integer after rounding instead of changing tests in valhalla/test/skadi_service.cc
+      writer(static_cast<int64_t>(std::round(height)));
     } else {
-      element->push_back({json::fixed_t{height, precision}});
+      writer(height);
     }
-    array->push_back(element);
+    writer.end_array();
     ++range;
   }
-  return array;
+  writer.end_array();
 }
 
-json::ArrayPtr serialize_height(const std::vector<double>& heights,
-                                const uint32_t precision,
-                                const double no_data_value) {
-  auto array = json::array({});
-
+void serialize_height(rapidjson::writer_wrapper_t& writer,
+                      const std::vector<double>& heights,
+                      const uint32_t precision,
+                      const double no_data_value) {
+  writer.start_array("height");
+  writer.set_precision(std::max(precision, 1u));
   for (const auto height : heights) {
     // add all heights's to an array
     if (height == no_data_value) {
-      array->push_back(nullptr);
+      writer(nullptr);
+    } else if (precision == 0) {
+      // cast to integer after rounding instead of changing tests in valhalla/test/skadi_service.cc
+      writer(static_cast<int64_t>(std::round(height)));
     } else {
-      array->push_back({json::fixed_t{height, precision}});
+      writer(height);
     }
   }
-
-  return array;
+  writer.end_array();
 }
 
-json::ArrayPtr serialize_shape(const google::protobuf::RepeatedPtrField<valhalla::Location>& shape) {
-  auto array = json::array({});
+void serialize_shape(rapidjson::writer_wrapper_t& writer,
+                     const google::protobuf::RepeatedPtrField<valhalla::Location>& shape) {
+  writer.start_array("shape");
+  writer.set_precision(6);
```

Comment:
I believe there was some constant for this in midgard? `kCoordinatePrecision` or so?